update-testsuite.sh

```diff
@@ -3,22 +3,125 @@ set -e
 set -u
 set -o pipefail
 
-spec="../spec"
+repos='spec threads simd exception-handling gc bulk-memory-operations tail-call nontrapping-float-to-int-conversions multi-value host-bindings sign-extension-ops mutable-global'
 
-pushd "$spec" >/dev/null
-    git checkout master
-    git fetch origin
-    git merge origin/master
-    id=$(git log --max-count=1 --pretty=oneline | sed -e 's/ .*//')
-popd >/dev/null
+log_and_run() {
+    echo ">>" $*
+    if ! $*; then
+        echo "sub-command failed: $*"
+        exit
+    fi
+}
 
-cp "$spec"/test/core/*.wast .
+try_log_and_run() {
+    echo ">>" $*
+    $*
+}
 
-for new in *.wast; do
-    git add "$new" || :
+pushdir() {
+    pushd $1 >/dev/null || exit
+}
+
+popdir() {
+    popd >/dev/null || exit
+}
+
+update_repo() {
+    local repo=$1
+    pushdir repos
+        if [ -d ${repo} ]; then
+            log_and_run git -C ${repo} fetch origin
+            log_and_run git -C ${repo} reset origin/master --hard
+        else
+            log_and_run git clone https://github.com/WebAssembly/${repo}
+        fi
+
+        # Add upstream spec as "spec" remote.
+        if [ "${repo}" != "spec" ]; then
+            pushdir ${repo}
+                if ! git remote | grep spec >/dev/null; then
+                    log_and_run git remote add spec https://github.com/WebAssembly/spec
+                fi
+
+                log_and_run git fetch spec
+            popdir
+        fi
+    popdir
+}
+
+merge_with_spec() {
+    local repo=$1
+
+    [ "${repo}" == "spec" ] && return
+
+    pushdir repos/${repo}
+        # Create and checkout "try-merge" branch.
+        if ! git branch | grep try-merge >/dev/null; then
+            log_and_run git branch try-merge origin/master
+        fi
+        log_and_run git checkout try-merge
+
+        # Attempt to merge with spec/master.
+        log_and_run git reset origin/master --hard
+        try_log_and_run git merge -q spec/master -m "merged"
+        if [ $? -ne 0 ]; then
+            git merge --abort
```

Comment:
Does this result in a message being printed about the merge failing? I agree with what you said elsewhere that it's better to ping the proposals to update, so it'd be useful if this script makes it obvious which proposals need to be pinged :-).

Reply:
Yes, and also it collects all the failed repos and displays them at the end too.

Reply:
Cool.